pkg/webhook/controlplane/ensurer.go

```diff
@@ -149,6 +148,133 @@ func keepNodeNetworkEnvVarIfPresentInOldDeployment(new, old *appsv1.Deployment,
 	}
 }
 
+// EnsureAdditionalProvisionUnits ensures that additional required system units are added, that are required during provisioning.
+func (e *ensurer) EnsureAdditionalProvisionUnits(ctx context.Context, gctx gcontext.GardenContext, new, _ *[]extensionsv1alpha1.Unit) error {
+
+	cluster, err := gctx.GetCluster(ctx)
+	if err != nil {
+		return err
+	}
+
+	for _, worker := range cluster.Shoot.Spec.Provider.Workers {
+		if worker.DataVolumes != nil {
+			switch worker.Machine.Image.Name {
+			case "flatcar":
+				extensionswebhook.AppendUniqueUnit(new, extensionsv1alpha1.Unit{
+					Name:    "lvm-setup.service",
+					Enable:  ptr.To(true),
+					Command: ptr.To(extensionsv1alpha1.CommandStart),
+					Content: ptr.To(`[Unit]
+        Description=LVM Setup
+        ConditionFirstBoot=yes
+        DefaultDependencies=no
+        Before=local-fs-pre.target
+        [Service]
+        Type=oneshot
+        Restart=on-failure
+        RemainAfterExit=yes
+        ExecStart=/opt/bin/lvm.sh
+        [Install]
+        WantedBy=multi-user.target
+`),
```

Comment:
might be simpler to read if we have that in a module scoped var

Reply:
it's just one time used right now. you think we should make it a var though just in case?

Reply:
I wasn't thinking about the reuse case but rather about the readability.
If we have everything on the module scope the identation can be like it would be in the file.

e.g.
```
var lvmSetup = `
[Unit]
        Description=LVM Setup
        ConditionFirstBoot=yes
        DefaultDependencies=no
        Before=local-fs-pre.target
        [Service]
        Type=oneshot
        Restart=on-failure
        RemainAfterExit=yes
        ExecStart=/opt/bin/lvm.sh
        [Install]
        WantedBy=multi-user.target
`),
```

Just a optional suggestion, will not block this PR because of this